edge-modules/TwinTester/TwinOperator.cs

```diff
@@ -0,0 +1,349 @@
+                Twin newTwin = await this.registryManager.UpdateTwinAsync(Settings.Current.DeviceId, Settings.Current.ModuleId, patch, this.twinState.TwinETag);
+                this.twinState.TwinETag = newTwin.ETag;
+            }
+            catch (Exception e)
+            {
+                Logger.LogInformation($"Failed call to remove successful desired property updates: {e}");
+            }
+        }
+
+        async Task ValidateReportedPropertyUpdates()
+        {
+            Twin receivedTwin;
+            try
+            {
+                receivedTwin = await this.registryManager.GetTwinAsync(Settings.Current.DeviceId, Settings.Current.ModuleId);
+                this.twinState.TwinETag = receivedTwin.ETag;
+            }
+            catch (Exception e)
+            {
+                Logger.LogInformation($"Failed call to registry manager get twin: {e}");
+                this.twinState.LastTimeOffline = DateTime.UtcNow;
+                return;
+            }
+
+            TwinCollection propertyUpdatesFromTwin = receivedTwin.Properties.Reported;
+            Dictionary<string, DateTime> reportedPropertiesUpdated = await this.storage.GetAllReportedPropertiesUpdated();
+            TwinCollection propertiesToRemoveFromTwin = new TwinCollection();
+            foreach (KeyValuePair<string, DateTime> reportedPropertyUpdate in reportedPropertiesUpdated)
+            {
+                string status;
+                if (propertyUpdatesFromTwin.Contains(reportedPropertyUpdate.Key))
+                {
+                    try
+                    {
+                        await this.storage.RemoveReportedPropertyUpdate(reportedPropertyUpdate.Key);
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.LogError($"Failed to remove validated reported property id {reportedPropertyUpdate.Key} from storage: {e}");
+                        continue;
+                    }
+
+                    status = $"{(int)StatusCode.Success}: Successfully validated reported property update";
+                    Logger.LogInformation(status + $" {reportedPropertyUpdate.Key}");
+                }
+                else if (this.IsPastFailureThreshold(reportedPropertyUpdate.Value))
+                {
+                    status = $"{(int)StatusCode.ReportedPropertyUpdateNotInCloudTwin}: Failure receiving reported property update";
+                    Logger.LogError(status + $" for reported property update {reportedPropertyUpdate.Key}");
+                }
+                else
+                {
+                    continue;
+                }
+
+                propertiesToRemoveFromTwin[reportedPropertyUpdate.Key] = null; // will later be serialized as a twin update
+                await this.CallAnalyzerToReportStatus(Settings.Current.ModuleId, status, string.Empty);
+            }
+
+            foreach (dynamic pair in propertiesToRemoveFromTwin)
```

Comment:
Potentially place in above loop.

Reply:
The issue is that if twin properties clear out we want to remove on the next validation cycle.